assets/src/edit-story/components/canvas/carousel/index.js

```diff
@@ -161,14 +199,23 @@ function Carousel() {
 
   const isAtBeginningOfList = 0 === scrollPercentage;
   const isAtEndOfList = 1 === scrollPercentage;
+  const scrollByPx = carouselSize.width;
+
+  const Item = isCompact ? CompactIndicator : DraggablePage;
+  const itemMarginRight = isCompact
+    ? CAROUSEL_COMPACT_RIGHT_MARGIN
+    : CAROUSEL_THUMB_RIGHT_MARGIN;
+  const [pageThumbWidth, pageThumbHeight] = calculatePageThumbSize(
+    carouselSize
+  );
```

Comment:
This could be memoized. Not super expensive, but still.

Reply:
I tend to not use `useMemo` if I'm not dealing with sorting/filtering or any _expensive calculations_. 
This topic is huge and probably we should measure before optimizing. Because I feel that in a case like this, we can end up wondering what is faster (in CPU): simple math (here) or branching (in useMemo) or memory impact of both solutions... what we have here is less code and that is _(usually)_ better (e.g. for reading).

I've seen other (bigger) places that could benefit from optimization. We can come back to this topic when the time is right (with all tests in place, measure, optimize, measure, decide).

Reply:
My main reasoning for memoizing is generally to avoid re-renders by not re-creating complex objects without needing to (rather than optimizing for speed), but as we're destructing the object here to simple values before using them as props for other components, I think we're good. If the thumb size stays the same, no needless re-rendering of thumbs will take place.